embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java

```diff
@@ -82,12 +90,361 @@ public PageOutput open(TaskSource taskSource, Schema inputSchema,
         String getRule();
     }
 
+    private interface CharacterTypesRule
+            extends Rule {
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("replace")
+        @ConfigDefault("\"_\"")
+        String getReplace();
+    }
+
+    private interface FirstCharacterTypesRule
+            extends Rule {
+        @Config("replace")
+        @ConfigDefault("null")
+        Optional<String> getReplace();
+
+        @Config("pass_types")
+        @ConfigDefault("[]")
+        List<String> getPassTypes();
+
+        @Config("pass_characters")
+        @ConfigDefault("\"\"")
+        String getPassCharacters();
+
+        @Config("prefix")
+        @ConfigDefault("null")
+        Optional<String> getPrefix();
+    }
+
+    private interface TruncateRule
+            extends Rule {
+        @Config("max_length")
+        @ConfigDefault("128")
```

Comment:
I put the comment https://github.com/embulk/embulk/pull/494#discussion_r85686826. I will try to fix it.